build_tools/travis/before_install.sh

```diff
@@ -35,43 +36,41 @@ if [[ $TRAVIS_OS_NAME == 'osx' ]]; then
 else
     # Install some custom requirements on Linux
     # g++4.8.1
-    if [ "$CXX" == "g++" ]; then sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test; fi
+    sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test; 
+    sudo apt-get install -qq g++-4.8;
+    export CXX="g++-4.8"; 
```

Comment:
why not protecting these lines in the else of the "if [ "$CXX" == "clang++" ]; then"